### PR TITLE
Adding horses betting system on hippodrome

### PR DESCRIPTION
**Pull request by dvincent56**
- Bet on horse race from hippodrome with your personal founds
- If you loose, you loose your bet
- If you win, you double your bet, and you quadruple your bet during festivals
- During the race, your bet amount is not accessible for emperor gift of city donation.
- A notification appears at the end of the race if your horse wins the race or not.

betting

### CODE REVIEW

**Comment by Riww** [comment]
How about 50% chance for green horse to win if spring (mar,apr,jun). The same for red/summer, yellow/autumn, and white/winter. The catch could be you never know exactly when the race will end. Limit max bet and and number of bets per year to avoid unlimited funds increase. Also, 3-times bet win during no-festival, as 2 times is too much rip-off.

**Comment by crudelios** [comment]
Can you fix the conflict so I can look at this, please? Thanks!

**Comment by dvincent56** [comment]
rebase done

File: src/city/race_bet.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                city_data.emperor.personal_savings = 0;
+            }
+            city_warning_show(WARNING_BET_DEFEAT);
+        }
+
+        // reset previous bet
+        city_data.games.bet_value = NO_BET;
+        city_data.games.bet_amount = 0;
+    }
+}
```
Comment (crudelios) [inline src/city/race_bet.c:37]: New line at end of file

File: src/window/race_bet.c [hunk 2]
```diff
@@ -0,0 +1,141 @@
+
```
Comment (crudelios) [inline src/window/race_bet.c:1]: No empty first line

File: src/window/race_bet.c [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+static void draw_foreground(void)
+{
+    graphics_in_dialog_with_size(16 * data.width_blocks, 16 * data.height_blocks);
+
+    int image_id = assets_get_image_id("UI_Elements", "Hipp_Blues_UH");
+    int image_id_focus = assets_get_image_id("UI_Elements", "Hipp_Blues_H");
+
+    for (int i = 0; i < 4; i++) {
+        int slice_id = (2 * i);
+        int id = data.focus_button_id == (i+1) || data.bet_value == (i+1) ? image_id_focus : image_id;
```
Comment (crudelios) [inline src/window/race_bet.c:90]:
```suggestion
        int id = data.focus_button_id == (i + 1) || data.bet_value == (i + 1) ? image_id_focus : image_id;
```

**Review by crudelios — CHANGES_REQUESTED** [review]
In terms of code, just a few style changes.

Is this still crashing though?

Also, maybe a "Bet" and "Cancel" buttons?

And you shouldn't be able to change a bet after you've set one and before the race ends.

**Comment by dvincent56** [comment]
>Also, maybe a "Bet" and "Cancel" buttons?
>And you shouldn't be able to change a bet after you've set one and before the race ends.

Agreed, I will add Bet button to confirm betting, probably not a cancel button because when you bet on race, you can't cancel your bet in real life. And asap I confirm bet, I can't change my choice.

**Pull request merged**